### PR TITLE
support sha256 for StreamDriverAuthMode="x509/fingerprint"

### PR DESCRIPTION
**Pull request by codemaker219**
Support sha256 hashes for "x509/fingerprint" for gtls and ossl
#4928

### CODE REVIEW

File: runtime/nsd_ossl.c [hunk 1]
```diff
@@ -633,6 +644,9 @@ osslChkPeerFingerprint(nsd_ossl_t *pThis, X509 *pCert)
 		if(!rsCStrSzStrCmp(pstrFingerprint, pPeer->pszID, strlen((char*) pPeer->pszID))) {
 			dbgprintf("osslChkPeerFingerprint: peer's certificate MATCH found: %s\n", pPeer->pszID);
 			bFoundPositiveMatch = 1;
+		} else if(!rsCStrSzStrCmp(pstrFingerprintSha256, pPeer->pszID, strlen((char*) pPeer->pszID))) {
+			dbgprintf("osslChkPeerFingerprint: peer's certificate MATCH found: %s\n", pPeer->pszID);
```
Comment (alorbach) [inline runtime/nsd_ossl.c]: Perhaps adding "Sha256" and "Sha1" to dbgprintf() ? would help to debug which matched the fingerprint

**Comment by alorbach** [comment]
Thanks for the contribution. The PR looks fine to me, the only recommendation I have is to add SHA1 / SHA256 to the dbgprintf outputs, so you can see which type matched.

**Comment by codemaker219** [comment]
Thanks for the feedback. I modified the log message for gtls and added them to ossl.

**Comment by alorbach** [comment]
Thanks for the changes :), I believe there is a limit of commits in one PR. Can you squash the last 3 commits into one? Thanks

**Comment by codemaker219** [comment]
Done

**Comment by rgerhards** [comment]
Thanks for the PR, but I think we need to update the doc a bit. Please add doc to https://github.com/rsyslog/rsyslog-doc/ If you think this is not necessary, please let me know why not.

**Comment by rgerhards** [comment]
> Should I update the documentation?

good catch! yes, please update!

**Comment by codemaker219** [comment]
https://github.com/rsyslog/rsyslog-doc/pull/980

**Pull request merged**